// WITH_STDLIB

const val IO_SIZE1 = 256u
const val IO_SIZE2 = <!CONST_VAL_WITH_NON_CONST_INITIALIZER!>256u * 1024u<!>
const val IO_SIZE3 = <!CONST_VAL_WITH_NON_CONST_INITIALIZER!>256u + 1024u<!>
const val IO_SIZE4 = <!CONST_VAL_WITH_NON_CONST_INITIALIZER!>256u - 1024u<!>
const val IO_SIZE5 = <!CONST_VAL_WITH_NON_CONST_INITIALIZER!>256u / 1024u<!>
const val IO_SIZE6 = <!CONST_VAL_WITH_NON_CONST_INITIALIZER!>256u % 1024u<!>
